### ISSUE TITLE
ConnectionResetError when using example

### ISSUE THREAD

**Issue by GlenWright** [No767/discord-ext-ipcx#30]
Hi

I'm trying to get simple IPC to work so that I can perform basic actions via an API like posting a message in a channel etc.

I'm currently using this library and the examples within without any modifications: https://github.com/No767/discord-ext-ipcx/tree/main/examples/basic-ipc

I keep getting `ConnectionResetError: Cannot write to closing transport` from webserver.py and can't seem to figure out why. I'm making the request via Postman to http://127.0.0.1:8000/ using the GET method.

Any help appreciated.

**Comment by No767** [comment]
Hi there. Haven't been able to take a look at this issue yet but will shortly once I have the time to diagnose the issue.

**Comment by No767** [comment]
I have figured out the cause. Simply to put it, the old code attempted to hold an persistent websocket connection, which is not recommended. Just like an HTTP request, websocket requests are closed once used. Thus, when the code tries to reuse the connection, it's closed and nowhere to be found. Thus, as it needs to write to the server for auth and ping reasons, it fails since there is no underlying connection to begin with.

This is now fixed by now using `aiohttp.ClientSession.ws_connect` as an context manager, thus when the code is done, it will automatically close down the connection. This approach is recommended in the examples found in the [aiohttp documentation](https://docs.aiohttp.org/en/stable/client_quickstart.html#websockets). The commit that fixes this also upgrades other misc parts of the client class
